frontend/src/views/Lobby.tsx

```diff
@@ -333,6 +338,40 @@ function LobbyPage() {
       });
   };
 
+  /**
+   * Update the list of selected problems
+   */
+  const updateSelectedProblems = (newProblems: SelectableProblem[]) => {
+    setError('');
+    setLoading(true);
+
+    const prevProblems = selectedProblems;
+    setSelectedProblems(newProblems);
+
+    const settings = {
+      initiator: currentUser!,
+      problems: newProblems,
+    };
+
+    updateRoomSettings(currentRoomId, settings)
```

Comment:
I think I'm missing something here; the object you pass in has `problems` which is a `SelectableProblem[]`, but the `settings` object for `updateRoomSettings` has no such field - how does this work?

Reply:
haha yeah i forgot to update the typescript type (mentioned in another comment). as for why it works, i think typescript allows you to have additional properties (which is why it allows `getAllProblems => Problem[]` to convert to `SelectableProblem[]`), so it just ignores the optional `problems` property while this correct one is sent anyways

Reply:
Got it, thanks for updating this 👍